File: 01_projects/code/java-selenium/cbaps-dawms/src/main/java/com/automation/cbaps/pages/RoutingApprovalPage.java
package com.automation.cbaps.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import com.automation.cbaps.library.GlobalSelenium;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class RoutingApprovalPage {
    private static final Logger log = LogManager.getLogger(RoutingApprovalPage.class);
    
    private WebDriver driver;
    private GlobalSelenium gs;
    
    private By approverDropdown = By.id("approver");
    private By submitButton = By.xpath("//button[text()='Submit for Approval']");
    
    public RoutingApprovalPage(WebDriver driver, GlobalSelenium gs) {
        this.driver = driver;
        this.gs = gs;
        gs.waitForElementVisibility(approverDropdown);
        log.info("✅ RoutingApprovalPage initialized");
    }
    
    public StatusTrackerPage submitForApproval(String approver) {
        log.info("✍️ Submitting for approval to: " + approver);
        gs.selectDropDown(approverDropdown, approver);
        gs.clickButton(submitButton);
        log.info("✅ Submitted for approval");
        return new StatusTrackerPage(driver, gs);
    }
}

File: 01_projects/code/python-selenium/cbaps-dawms/cbaps/pages/routing_approval_page.py
from selenium.webdriver.common.by import By
from loguru import logger

class RoutingApprovalPage:
    def __init__(self, driver, selenium):
        self.driver = driver
        self.selenium = selenium
        self.approver_dropdown = (By.ID, "approver")
        self.submit_btn = (By.XPATH, "//button[contains(text(), 'Submit for Approval')]")
        self.selenium.wait_for_element_visible(self.approver_dropdown)
    
    def submit_for_approval(self, approver: str):
        from .status_tracker_page import StatusTrackerPage
        logger.info(f"✍️ Submitting: {approver}")
        self.selenium.select_dropdown(self.approver_dropdown, approver)
        self.selenium.click_element(self.submit_btn)
        return StatusTrackerPage(self.driver, self.selenium)
